nessus-agent/elx/install.sls

```diff
@@ -0,0 +1,65 @@
+#
+# This salt state installs the Nessus Agent using the provided
+# URL to the rpm source file.  Once installed, the agent will be
+# configured and linked to the Nessus server using the provided
+# parameters (Nessus 64 hexadecimal-digit key, Nessus server FQDN and
+# port, Nessus Agent Group Name).  The final step is to start tHe
+# Nessus Agent service.
+#################################################################
+{%- from tpldir ~ '/map.jinja' import nessus with context %}
+
+{%- for log_file,log_rotation in nessus.log_config.items() %}
+  {%- for rotation_param,rotation_value in log_rotation.items() %}
+Add Log Config {{ log_file }} {{ rotation_param }}:
+  file.replace:
+    - name: /opt/nessus_agent/var/nessus/log.json
+    - pattern: '"file": "{{ log_file }}"'
+    - repl: '"{{ rotation_param }}": "{{ rotation_value }}",\n                \g<0>'
+    - watch:
+      - cmd: Pause For Log File
+  {%- endfor %}
+{%- endfor %}
+
+Create Sym-link To Log Dir:
+  file.symlink:
+    - name: /opt/nessus_agent/var/nessus/logs
+    - target: /var/log/nessus/logs
+    - user: root
+    - group: root
+    - mode: 0755
+    - makedirs: True
+    - require:
+      - file: Pre-Create Nessus Log Directory
+
+Enable Nessus Agent:
```

Comment:
I _think_ this state can be removed, as this state is no longer dealing with the service. With this state present, when paired with `.configure` which ensures the service is running, this will result in constant changes every execution. The service will stop, then start again, every time. Or perhaps this needs to be `service.running` instead of `service.dead` (considering the name of the state is "Enable Nessus Agent", and the log file dependency on this state)

Reply:
The "enable" is to reflect that it's simply ensuring that, in case the installation RPM's `%post` includes a start action, this stanza will put the service into a "ready for configuration" state. I don't _know_ that it's needed, just hedging.

Reply:
I kinda get that, but `service.dead` will stop the service. And then later `service.running` will start the service. And so we have persistent changes from run to run, which means it is no longer stateful. I'd rather only do that if we actually have a problem, and so far we have not.

Reply:
Okie doke. Like I say, was hedging – which I tend to do when I'm dealing with what amounts to black box (prior this morning's attempts to validate that the modularized content works equivalently to the pre-modular code, I've lacked good access to the nessus software to observe an end-to-end installation). I'll yank it. Can add it back if it turns out to need to be there.

Reply:
Removed